src/transformStateWithClones.js

```diff
@@ -7,7 +7,35 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let copiedState = { ...state };
+  const stateHistory = [];
+
+  for (const action of actions) {
+    const { type, extraData, keysToRemove } = action;
+
+    switch (type) {
+      case 'addProperties':
+        Object.assign(copiedState, extraData);
+        break;
+
+      case 'removeProperties':
+        for (const key of keysToRemove) {
+          delete copiedState[key];
+        }
+        break;
+
+      case 'clear':
+        copiedState = {};
+        break;
+
+      default:
+        return state;
```

Comment:
[CODE STYLE]: The default case in the switch statement should not return the original state object. Instead, it should handle unexpected action types gracefully, perhaps by logging an error or ignoring the action. Returning the original state might be misleading as it stops the entire transformation process unexpectedly.